### ISSUE TITLE
Empty response[Object] `onFbLogin(...)` on page load and logged in user

### ISSUE THREAD

**Issue by desert** [adi518/vue-facebook-login-component#43]
This causes the @login to trigger with and empty object. Should send the same object in order to avoid developers to handle two cases. Also documentation show it should send the object.

**Comment by desert** [comment]
See #41

**Comment by desert** [comment]
If you push the login button you get the object as parameter but if you refresh the page the function gets called with empty parameters. However FB SDK does deliver the object. It is just this line:

https://github.com/adi518/vue-facebook-login-component/blob/f42d8cfce96feea3b2c3e19952ce972533cb2066/packages/vue-facebook-login-component/src/components/Scope.js#L50

**Comment by adi518** [comment]
Thanks, that's the result of code duplication. Working on a fix, thanks.

**Comment by desert** [comment]
As a workaroud, I'm catching `@sdk-init` and calling FB SDK to catch a if user is already logged in on a page refresh.

**Comment by desert** [comment]
That has the unfortunate side effect of duplicating the `getLoginStatus()` call , as it has been already called in `async created()`.

**Comment by adi518** [comment]
The login event that fires without payload means the user is logged in, not a login attempt that either succeeded or failed. It means, you can be sure the user is logged-in/connected. Currently, you can tell initial login from user-initiated login by looking at the payload, if it's empty, it's the initial login, which is actually just login status coming back with a `connected` status response.

**Comment by desert** [comment]
Yes but I need the payload to synchronize the login status to my own REST API. I can sync on `@login` and get my own API token, which I save. 

But the edge case is where some user deleted, flushed parts of the localstorage or even hacked it. SDK preserves the data but I need to revive it, however I know it is logged in but loose the token, user id and the full response from FB. So I need to re-sync manually. 

As I see it, you already have it on the response, so why not identify it with a third paramenter `login(fbresponse, initialLogin:true)` so we know if it is an initial login or a re-login, that would also keep the api unchanged for legacy code.

**Comment by adi518** [comment]
What do you mean "third parameter"? where? I didn't see such flag in Facebook docs.

**Comment by desert** [comment]
Not FB, but the component. As FB does they always provide the response, no third parameters. I think the component should always provide the paramenter too. And if you want to differentiate then send `initialLogin: true/false`

**Comment by desert** [comment]
In short, 

`this.$emit({ ...response, initial: true })`

just as you said

**Comment by adi518** [comment]
Right, we can solve it like that. The downside is essentially mixing together the response and our own properties, but it's probably not severe enough to reconsider.

**Comment by M1CK431** [comment]
@adi518 My personal feeling is: if someone is using the initial `login` event, he can't use the payload which make this event useless. In addition, the documentation describe the `login` event as "User logged in." and so I think that no one is expecting to have this event without user interaction (= initial state)

I think the best solution is to emit a new event (named `init` for example) with the initial login state as payload. The `login` event will be then "reserved" for it's original purpose: get the result after a user login interaction.

I agree that it could be a breaking change however but it's not so hard to handle it with just one line in README for the release of v2.3.0.

**Comment by desert** [comment]
Both the "initial" login `res` payload and the re-login `res` should be identical in order to sync any backends without data reformatting. 

But I _think_, the FB payload has that info, not sure tho. Maybe the `initial:true` is in fact not needed.

**Comment by desert** [comment]
FB docs require you to re-check the token on a backend server in order to convert it to a long living token (for Graph API and other stuff). So we are forced to send it to backend.

My code is basically this:

```
onSDKInit(FB){
   this.ReCheckLogin(FB)
   .then((res) => {
       registerMyOwnSession(myApiPrivateURL, res)
   })
}
onFbLogin(res){
  if (res)
     registerMyOwnSession(myApiPrivateURL, res)
}
```

If  `res` is identical, as FB SDK already does, then it should be a lot simpler, single liner:
```
onFbLogin( (res) =>  registerMyOwnSession(myApiPrivateURL, res) )
```

And both of next would happen:

1. We avoid duplicate SDK calls: `FB.getLoginStatus()`
2. `ReCheckLogin(FB)` could be entirely removed

**Comment by M1CK431** [comment]
@desert as you explain, FB require to send the sort lived token (the one we get from front end side) to a backend which have to "transform" it to a long lived token. Of course, this action has to be done on user login (after the user interaction). And so why do you need to resend a short lived token using an `init` event? Why the backend isn't responsible to renew the long lived token by itself?

**Comment by desert** [comment]
> why do you need to resend a short lived token using an init event?

I dont.

**Comment by desert** [comment]
I would only need this line 
https://github.com/adi518/vue-facebook-login-component/blob/f42d8cfce96feea3b2c3e19952ce972533cb2066/packages/vue-facebook-login-component/src/components/Scope.js#L50 

To be exactly as this:
https://github.com/adi518/vue-facebook-login-component/blob/f42d8cfce96feea3b2c3e19952ce972533cb2066/packages/vue-facebook-login-component/src/components/Scope.js#L98

**Comment by adi518** [comment]
> @adi518 My personal feeling is: if someone is using the initial `login` event, he can't use the payload which make this event useless.

I think we argue semantics here. You argue that login event should emit only if it is initiated by the user, and I argue that it should also emit when we check the login status and find out the user is still logged in. I get your point, but it's a little difficult to pick which is best TBH. I'm tempted to just rename the initial `login` event to `login-status`, but that would mean I'll have to add another event, possibly `initial-login` or `initial-connect`. In the past I emitted a login event whether it ended in success or failure, and another event `connect` on every successful login (whether initiated by the user or not, like now, you couldn't really tell). I changed that in version 2 in hopes to simplify things. In version 2 login event only emits if it was successful. You can see version one [here](https://github.com/adi518/vue-facebook-login-component/blob/f65af855ec772d24d3c350bcc42a4e0520e2da66/src/components/FBLogin.Scope.js).

In attempt to conclude this issue, I think I may restore login emit on both success/failure, add another event `connect` when it was successful, event `initial-connect` whether login-status ended with "connected" status and perhaps another event `login-status` wit a response payload, whether user is connected or not.

**Comment by adi518** [comment]
> @adi518 My personal feeling is: if someone is using the initial `login` event, he can't use the payload which make this event useless.

You can tell initial login from user-initiated login by looking at the payload, you said it yourself, so it can be useful. I do agree the payload should be there, but then we won't be able to tell initial this event from user-initiated login events, unless I add an `initial: true` flag.

We really argue semantics here. You argue that login event should emit only if it is initiated by the user, and I argue that it should also emit when we check the login status and find out the user is still logged in. I get your point, but it's a little difficult to pick which is best TBH. I'm tempted to just rename the initial `login` event to `initial-connect`, add the response payload, and add another event `login-status`.

In the past I emitted a login event whether it ended in success or failure, and another event `connect` on every successful login (whether initiated by the user or not, like now, you couldn't really tell). I changed that in version 2 in hopes to simplify things. In version 2, login event only emits if it was successful. You can see latest version 1 source [here](https://github.com/adi518/vue-facebook-login-component/blob/f65af855ec772d24d3c350bcc42a4e0520e2da66/src/components/FBLogin.Scope.js).

In attempt to conclude this issue, one solution is to restore login emit on both success/failure, add another event `connect` when it was successful, an `initial-connect` event when login-status ended with "connected" status and perhaps another event `login-status` with the response payload, whether user is connected or not. The downside is breaking stuff.

**Comment by desert** [comment]
If there is no payload, everybody needs to do a double request in order to get user info. So beyond semantics, there is pragmatism, and sending the payload `this.$emit('login', status)` just saves time.

**Comment by M1CK431** [comment]
> You can tell initial login from user-initiated login by looking at the payload, you said it yourself, so it can be useful.

@adi518 That's not exactly that. I explain that I'm using the lake of payload of the initial `login` event to **ignore** it (= useless and even unexpected event).

There is a way to deal with this issue without breaking anything:
- keep the `login` as is (no breaking change)
- add an `initial-status` with the `status` as payload

This way @desert will be able to get the login and the initial status easily.

**Comment by adi518** [comment]
@desert Your suggestion creates is a bit confusing. Do you mean just the "status" as a payload or the entire response? the [response](https://developers.facebook.com/docs/reference/javascript/FB.getLoginStatus) is an object that can include more than just the status property.

I'm against keeping useless/confusing stuff. Some action items:

* Remove `login` event from `create` lifecycle (we'll add a line in docs about this).
* Add `initial-connect` event with  `response` payload.
* Add `login-status` event with `response` payload.

This way you get everything.

> @adi518 That's not exactly that. I explain that I'm using the lake of payload of the initial `login` event to **ignore** it (= useless and even unexpected event).

Did you mean "lack"?

**Comment by desert** [comment]
I think you are overthinking this. Just do what FB does, and send the `res`, this component is just hiding it,  I don't see why it should hide it. 

`this.$emit('login', status) `

**Comment by adi518** [comment]
Sorry, I was a bit busy to get done with this issue. I plan to follow the action items listed above.

**Comment by desert** [comment]
I was getting this error 
![image](https://user-images.githubusercontent.com/723830/82762900-45027e00-9dc9-11ea-97f1-b1aab3b4c49a.png)

Just by letting the tab idle, so I did a bit of research. I think you should read willsmanley's answer here:

https://stackoverflow.com/a/58193545/2573334

**Comment by adi518** [comment]
Thanks, looks like an easy fix. 👍

**Comment by desert** [comment]
Just pinging to see if we can receive the response. 


> vue-facebook-login-component/packages/vue-facebook-login-component/src/components/Scope.js
